src/core/io/platform/f3xx/ADCf3xx.cpp

```diff
@@ -38,6 +38,7 @@ ADCf3xx::ADCf3xx(Pin pin) : ADC(pin) {
 
     // Maximum number of ADC channels have already been added
     if (rank == MAX_CHANNELS) {
+        log::LOGGER.log(log::Logger::LogLevel::WARNING, "ADC1 ALREADY HAS MAX NUMBER OF CHANNELS!!");
```

Comment:
Should be an error since the class is not fully initialized.

Reply:
What do you mean by the class is not fully initialized? the rank is only max if there are already 15 channels open on the ADC.

Reply:
I think he means that this instance of the class won't be fully initialized, so it's not going to behave properly. Therefore, you should report an error instead of a warning